lib/pure/os.nim

```diff
@@ -1688,13 +1688,28 @@ proc getFileInfo*(path: string, followSymlink = true): FileInfo =
     rawToFormalFileInfo(rawInfo, path, result)
 
 proc isHidden*(path: string): bool =
-  ## Determines whether a given path is hidden or not. Returns false if the
-  ## file doesn't exist. The given path must be accessible from the current
-  ## working directory of the program.
-  ##
-  ## On Windows, a file is hidden if the file's 'hidden' attribute is set.
-  ## On Unix-like systems, a file is hidden if it starts with a '.' (period)
-  ## and is not *just* '.' or '..' ' ."
+  ## Determines whether a given path is hidden or not.
+  ## On Windows, a path is hidden if it exists and it's 'hidden' attribute is set.
+  ## On posix, a hidden path starts with `.` (see examples).
+  ## reference: https://en.wikipedia.org/wiki/Hidden_file_and_hidden_directory
+  ## NOTE: on OSX, in addition to the "dotfile" behaviour, existing paths with the
+  ## "Invisible" attribute are hidden in Finder, although not in ls
+  ## NOTE: the behavior of certain corner cases (see examples) ".foo/" is
+  ## unspecified, subject to change (can be avoided using `normalizePath`)
+  runnableExamples:
+    when defined(posix):
+      doAssert ".foo".isHidden
+      doAssert ".foo.ext".isHidden
+      doAssert "bar/.foo.ext".isHidden
+      doAssert(not "bar".isHidden)
+      doAssert(not ".".isHidden)
+      doAssert(not "..".isHidden)
+      doAssert(not ".foo/bar".isHidden)
+      doAssert(not "".isHidden)
+      # TODO: decide on these:
+      # doAssert ".foo/".isHidden # un-specified
+      # doAssert ".foo/.".isHidden # un-specified
```

Comment:
Don't abuse ``runnableExamples`` for a test suite. The docs should show one or two examples, not an exhaustive list of checks.

Reply:
done, kept the minimum sufficient set to clarify semantics